### PR TITLE
Adapt to Qt6

### PR DESCRIPTION
**Pull request by dschulz**
This is a patch to have qt-nats play nicely with Qt6. Tested locally with trivial examples, TLS works OK. 
Needs some review and testing.

### CODE REVIEW

**Comment by pkoretic** [comment]
Thanks @dschulz, much appreciated, let me do some tests and I'll merge it.

File: README.md [hunk 1]
```diff
@@ -1,13 +1,48 @@
...
+A [Qt](https://www.qt.io) C++11 client for the [NATS messaging system](https://nats.io).
 
 [![License MIT](https://cdn.rawgit.com/pkoretic/qt-nats/badges/license.svg)](http://opensource.org/licenses/MIT)
 [![Language (C++)](https://cdn.rawgit.com/pkoretic/qt-nats/badges/powered_by-C%2B%2B-blue.svg)](http://en.cppreference.com/w/cpp/language)
 
 ## Installation
 
-This is a header-only library that depends on Qt5. All you have to do is include it inside your
-project. It depends on the Qt `network` module so don't forget to add it to your `project-name.pro` file (```QT += network```).
+*NB: This specific branch was adapted to work with Qt 6.*
```
Comment (pkoretic) [inline README.md]: this is not needed, it will be released as a new major library version

File: natsclient.h [hunk 2]
```diff
@@ -10,9 +10,8 @@
 #include <QSslSocket>
 #include <QStringBuilder>
 #include <QUuid>
-#include <QTextCodec>
 
-#include <memory>
+//#include <memory>
```
Comment (pkoretic) [inline natsclient.h:14]: should be removed

**Comment by pkoretic** [comment]
Thanks @dschulz, it took a bit but finally merged!

**Comment by dschulz** [comment]
Awesome! Thanks @pkoretic !

**Pull request merged**